src/modules/todos.js

```diff
@@ -86,27 +86,35 @@ const renderTodoList = () => {
       }
     }
 
-    function editTodo(todo) {
-      const todoIndex = todo.index;
-      input.value = todo.description;
-      addItemButton.innerText = 'Save';
-
-      const saveItem = () => {
-        todos[todoIndex].description = input.value;
-        localStorage.setItem('todos', JSON.stringify(todos));
-        renderTodoList();
-        hideOptionsMenu();
-        addItemButton.innerText = '+';
-        addItemButton.classList.add('plus-on-edit');
-        addItemButton.removeEventListener('click', saveItem);
-      };
+    function editTodo(todo, index) {
+      if (todo !== undefined) {
+        input.value = todo.description;
+
+        const saveItem = () => {
+          if (input.value !== '') {
+            // Updating the description of the existing todo at index
+            todos[index].description = input.value;
+          } else {
+            // Here I remove the todo at index
+            todos.splice(index, 1);
+          }
+
+          localStorage.setItem('todos', JSON.stringify(todos));
+          renderTodoList();
+          hideOptionsMenu();
+          addItemButton.innerText = '+';
+          addItemButton.classList.add('plus-on-edit');
+          addItemButton.removeEventListener('click', saveItem);
+        };
+
+        addItemButton.addEventListener('click', saveItem);
+        todosListItem.classList.remove('edited');
+      } else {
 
-      addItemButton.addEventListener('click', saveItem);
+      }
     }
 
-    const deleteTodo = (todo) => {
-      const todoIndex = todos.indexOf(todo);
-      todos.splice(todoIndex, 1);
+    const deleteTodo = () => {
       localStorage.setItem('todos', JSON.stringify(todos));
       renderTodoList();
     };
```

Comment:
[OPTIONAL] This requirement is optional but recommended so kindly don't skip it your delete functionality is not working so kindly fix your delete functionality :+1: